### DISCUSSION TITLE
For react-carousel, how can i update it so it's sorted/ordered by a new column i added?

### DISCUSSION THREAD

**Discussion by ghost** [pnp/sp-dev-fx-webparts#1695] [Q&A] [answered]
Hello, 

I have been updating 'react-carousel' webpart to accommodate several requests i have been receiving. 

One last remaining thing is to sort and display images by a new column 'Order' i added to each file(pictures). However, it just continues to order it by 'Id' instead of 'Order'...

```
  public async getImages(siteUrl: string, listId: string, numberImages: number): Promise<any[]> {
    let results: any[] = [];
    try {

      const web = new Web(siteUrl);
      results = await web.lists
        .getById(listId).items
        .select('Title','Description','File_x0020_Type', 'FileSystemObjectType','File/Name', 'File/ServerRelativeUrl', 'File/Title', 'File/Id', 'File/TimeLastModified', )
        .top(numberImages)
        .expand('File')
        .filter((`File_x0020_Type eq  'jpg' or File_x0020_Type eq  'png' or  File_x0020_Type eq  'jpeg'  or  File_x0020_Type eq  'gif' or  File_x0020_Type eq  'mp4'`))
        .orderBy('Order') // HERE!!! I changed from the default 'Id' to 'Order' yet nothing changes.  
        .usingCaching()
        .get();
    } catch (error) {
      return Promise.reject(error);
    }
    // sort by name



    return results;
  }
```

Could someone please guide me to how i can troubleshoot this issue? I have been stuck on this "simple" problem for hours after hours, and can't get myself to resolve this. 

Thank you very much in advance

**Comment by AriGunawan** [comment, marked as answer]
Hi @edwardNuvision,
Have you found the solution to your issue?
It looks like `Order` is one of the default fields of a Document Library item.
Please check again the internal name of your new column. I'm guessing it might be `Order0` because `Order` already exists.
Then try to use the actual internal name in your code. It should be working as expected.
